analyses/molecular-subtyping-pathology/pathology_free_text-subtyping-jxg.Rmd

```diff
@@ -0,0 +1,124 @@
+library(tidyverse)
+```
+
+### Input
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+results_dir <- file.path("results")
+histologies_file <- file.path(data_dir, "pbta-histologies-base.tsv")
+
+```
+
+### Output
+
+```{r}
+results_dir <- "results"
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+output_file <- file.path(results_dir, "juvenile-xanthogranuloma-subtypes.tsv")
+```
+
+## Read in data
+
+```{r}
+histologies_df <- readr::read_tsv(histologies_file, guess_max = 10000)
+```
+
+### Display `pathology_free_text_diagnosis` values
+
+```{r}
+histologies_df %>%
+  dplyr::filter(pathology_diagnosis == "Other") %>%
+  dplyr::filter(grepl("xanthogranuloma|jxg", pathology_free_text_diagnosis)) %>%
+  dplyr::select(pathology_free_text_diagnosis) %>%
+  dplyr::distinct()
+```
+
+## Recoding `harmonized_diagnosis` 
+
+We use the following table from [#998](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/998) to recode the `harmonized_diagnosis` values for samples with a `pathology_diagnosis` of Juvenile xanthogranuloma:
+
+pathology_diagnosis | subtyping module | pathology_free_text_diagnosis | broad_histology | short_histology | harmonized_diagnosis
+-- | -- | -- | -- | -- | --
+Other | NA | if contains "xanthogranuloma" or "jxg" | Histiocytic tumor | JXG | Juvenile xanthogranuloma
+
+
+```{r}
+jxg_df <- histologies_df %>%
+  # We are only concerned with samples where the pathology_free_text_diagnosis
+  # contains xanthogranuloma|jxg
+  dplyr::filter(pathology_diagnosis == "Other") %>%
+  dplyr::filter(grepl("xanthogranuloma|jxg", pathology_free_text_diagnosis)) %>%
+  # Retain only relevant identifier and disease label columns
+  dplyr::select(Kids_First_Biospecimen_ID,
+                Kids_First_Participant_ID,
+                sample_id,
+                tumor_descriptor,
+                pathology_free_text_diagnosis,
+                broad_histology,
+                short_histology) %>%
+  # To smooth the way for string detection for the pathology free text, we
+  # add a column where all of the text is converted to lowercase
+  dplyr::mutate(pathology_free_text_dx_lower = 
+                  stringr::str_to_lower(pathology_free_text_diagnosis)) %>%
+  # String detection in pathology free text per the table
+  dplyr::mutate(
+    harmonized_diagnosis = dplyr::case_when(
+      stringr::str_detect(pathology_free_text_dx_lower,
+                          "xanthogranuloma|jxg") ~ "Juvenile xanthogranuloma"
+    ),
+    broad_histology = dplyr::case_when(
+      stringr::str_detect(pathology_free_text_dx_lower,
+                          "xanthogranuloma|jxg") ~ "Histiocytic tumor",
+      TRUE ~ broad_histology
+    ),
+    short_histology = dplyr::case_when(
+      stringr::str_detect(pathology_free_text_dx_lower,
+                          "xanthogranuloma|jxg") ~ "JXG",
+      TRUE ~ short_histology
+    ),
+    Notes = "Updated via OpenPBTA subtyping from pathology_free_text_diagnosis"
+  ) %>%
```

Comment:
I'm going to merge this as is because I believe it to be correct, but for future reference I believe you could simplify this by moving the `str_to_lower()` step earlier, filtering like you do in 75 (now with the added "safety" of everything being lowercase), and then doing the following for the `mutate()` step:
```R
  # String detection in pathology free text per the table
  dplyr::mutate(
    harmonized_diagnosis =  "Juvenile xanthogranuloma"
    broad_histology = "Histiocytic tumor",
    short_histology = "JXG",
    Notes = "Updated via OpenPBTA subtyping from pathology_free_text_diagnosis"
  )
```

The idea being that you would not have any samples that didn't meet the  `stringr::str_detect(pathology_free_text_dx_lower,
                          "xanthogranuloma|jxg")` conditions in your data frame so you don't need the `case_when()`. (All of this is untested but I do something similar elsewhere...)

Reply:
Ah right makes sense, that would have been easier!